### DISCUSSION TITLE
Diff commands `jj diff`, `jj diffedit`, `jj interdiff` with `--from` argument are confusing in some cases

### DISCUSSION THREAD

**Discussion by zummenix** [jj-vcs/jj#3134] [General]
I don't look diffs for a range of revisions too often but when I do I always think twice what revision to pass in `--from` argument. I think the confusion comes from the fact that `--from` doesn't include the revision in the diff. For example: `jj diff --from u --to t`  shows the diff between `u` and `t` not including `u`. But intuitively I'd like to see the diff including `u`. On the other hand, when I look at a branch diff using  for example `jj diff --from main` I don't want to see changes of a revision that `main` branch is set to and this is intuitively what I want and how it currently behaves.

At least, I think there should be a clarification in the cli docs that `--from` doesn't include the revision in the diff for `jj diff`, `jj diffedit`, `jj interdiff` commands. Alternatively, I'd like to suggest ditching the `--from` argument completely or changing its semantics.

**Comment by martinvonz** [comment]
Note that the command works with any two commits, not only where `u` is an ancestor of `t`. Let's say you have this history:

```
B C
|/
A
```

If you now do `jj diff --from B --to C`, I would not expect that to show the diff from `B`'s parent (i.e. `A`) to `C`.

**Reply by zummenix** [reply]
> If you now do jj diff --from B --to C, I would not expect that to show the diff from B's parent (i.e. A) to C.

Yes, I have the same expectations.

So, the intuition behind `--from` and `--to` arguments can be the following: what changes need to be made starting from a revision (but not including) in `--from` that would lead to a state in `--to` revision. Is this a correct mental model in general that I can use?

**Reply by martinvonz** [reply]
Yes, that's correct.

**Comment by arxanas** [comment]
I think this matches the `git diff` semantics. It may be gitbrainedness for me, but checking the contents after `u` personally seems logical.

With your suggestion, it becomes ambiguous on how to include the changes "before" `u` when `u` is a merge commit and has multiple parents — which one would you use?

**Reply by zummenix** [reply]
I guess, my problem is that I have always thought about ranges of commits in context of a diff command. And arguments like `--from` and `--to` throw me away. If I try to think about them as "base" and "destination" (sort of) it becomes more clear for me.
